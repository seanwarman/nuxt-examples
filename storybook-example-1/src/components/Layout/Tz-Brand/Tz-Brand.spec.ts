import { shallowMount, VueWrapper } from '@vue/test-utils';
import Brand from './Tz-Brand.vue';
import { describe, beforeEach, it, expect } from 'vitest';

let wrapper: VueWrapper;

describe('Brand.vue setup', () => {
  beforeEach(() => {
    wrapper = shallowMount(Brand);
  });
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('sets a base class', () => {
    expect(wrapper.classes()).toContain('tz-brand');
  });
  it('img element has all the proper attributes', () => {
    const { alt, title, src } = wrapper.attributes();

    expect(src).equals('/src/assets/tredz-brand.svg');
    expect(title).equals('brand');
    expect(alt).equals('brand');
  });
});
