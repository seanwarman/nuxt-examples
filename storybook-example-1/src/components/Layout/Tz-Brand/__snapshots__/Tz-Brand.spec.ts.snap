// Vitest Snapshot v1

exports[`Brand.vue setup > renders correctly 1`] = `
<img
  alt="brand"
  class="tz-brand w-[91px] md:w-[102px]"
  src="/src/assets/tredz-brand.svg"
  title="brand"
/>
`;
